ksql-engine/src/main/java/io/confluent/ksql/util/ReferentialIntegrityTable.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.util;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+public class ReferentialIntegrityTable {
+
+  private final Map<String, ReferentialIntegrityTableEntry> referentialIntegrityTable;
+
+  public ReferentialIntegrityTable() {
+    referentialIntegrityTable = new HashMap<String, ReferentialIntegrityTableEntry>();
+  }
+
+  public void addSourceNames(Set<String> sourceNames, String queryId) {
```

Comment:
could change this to:
```
for (String sinkName: sinkNames) {
      referentialIntegrityTable.compute(sinkName, (key, value) -> {
        final ReferentialIntegrityTableEntry entry = value == null ? new
                ReferentialIntegrityTableEntry(sinkName) : value;
        entry.getSourceForQueries().add(queryId);
        return entry;
      });
    }
```

Similarly with the next method

Reply:
Refactored this after moving it to the metastore.